### PR TITLE
Change to not use persistent connections by default closes #2491

### LINKED ISSUE #2491: Fix issues with util.HTTPClient
First, pretty much every time this is used it's intended to use the localhost proxy -- we should make sure it always is. Otherwise nothing will be tunneled using domain fronting as intended. Second, most callers aren't explicitly closing connections, so it can easily lead to connection leaks as is for that reason alone. Beyond that though, it doesn't use an connection, read, or TLS handshake timeouts, so can lead to connection leaks as a result of that as well.

### PR DESCRIPTION
**Pull request by myleshorton**
Also added timeouts to the transport.

### CODE REVIEW

**Comment by myleshorton** [comment]
You guys mind taking a peek at this @ffw, @aranhoide, or @atavism? Seems like a pretty clear possible source of connection leaks otherwise.

**Comment by myleshorton** [comment]
@fffw can you look at this one?

File: src/github.com/getlantern/flashlight/util/http.go [hunk 1]
```diff
@@ -6,15 +6,53 @@ import (
...
+		TLSHandshakeTimeout: 10 * time.Second,
+
+		// This method is typically used for creating a one-off HTTP client
+		// that we don't want to keep around for future calls, making
+		// persistent connections a potential source of file descriptor
+		// leaks. Note the name of this variable is misleading -- it would
+		// be clearer to call it DisablePersistentConnections -- i.e. it has
+		// nothing to do with TCP keep alives along the lines of the KeepAlive
+		// variable in net.Dialer.
+		DisableKeepAlives: persistent,
```
Comment (fffw) [inline src/github.com/getlantern/flashlight/util/http.go]: From https://golang.org/src/net/http/transport.go#L361 , if persistent is true, `http.Transport` just closes the connection when it's idle. Should we use `!persistent` instead?

Comment (myleshorton) [inline src/github.com/getlantern/flashlight/util/http.go]: Great catch @fffw! For some reason I don't seem to get these messages from you -- probably something off with my email filters, so I'm just seeing this now. Fixing.

**Comment by myleshorton** [comment]
OK guys I'm going to merge this because otherwise we'll start to experience conflicts with new changes coming down the pipeline.

**Pull request merged**